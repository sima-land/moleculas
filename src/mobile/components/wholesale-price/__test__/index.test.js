import React from 'react';
import { mount } from 'enzyme';
import { WholesalePrice } from '../index';

describe('<WholesalePrice />', () => {
  it('should render without props', () => {
    const wrapper = mount(<WholesalePrice />);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should render with props', () => {
    const wrapper = mount(
      <WholesalePrice
        price={222222}
        description='<b>lorem</b>'
        text='Крупный опт «Галантерея и швейная галантерея»'
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should onDetailsClick calls on click', () => {
    const wrapper = mount(
      <WholesalePrice
        price={222222}
        description='lorem'
        onDetailsClick={jest.fn()}
        itemId={123456}
        text='Детские товары'
      />
    );
    wrapper.find('[role="button"]').simulate('click');
    expect(wrapper.props().onDetailsClick).toHaveBeenCalledTimes(1);
    expect(wrapper.props().onDetailsClick).toHaveBeenCalledWith(123456);
  });
});
