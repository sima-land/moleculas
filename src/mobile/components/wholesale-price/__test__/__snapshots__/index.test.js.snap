// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WholesalePrice /> should render with props 1`] = `
<WholesalePrice
  description="<b>lorem</b>"
  price={222222}
  text="Крупный опт «Галантерея и швейная галантерея»"
>
  <div
    className=""
  >
    <div
      className="wrapper"
    >
      <ForwardRef(Text)
        color="gray38"
        lineHeight={16}
        size={12}
        weight={600}
      >
        <span
          className="size-12 line-height-16 color__gray38 weight-600"
        >
          <Price
            currencyGraphemeClass="currency-grapheme"
            value={222222}
          >
            <span
              className="root"
            >
              222 222
            </span>
          </Price>
           / 
          <span
            className="wholesale-word"
          >
            Опт 
          </span>
          <WholesaleText
            className="wholesale-text"
            text="Крупный опт «Галантерея и швейная галантерея»"
          >
            <span
              className="wholesale-text"
            >
              Крупный опт «Галантерея и швейная галантерея»
            </span>
          </WholesaleText>
        </span>
      </ForwardRef(Text)>
    </div>
    <div
      className="icon-wrapper"
    >
      <span
        aria-label="Подробнее"
        className="icon"
        onClick={[Function]}
        role="button"
      >
        <InformationSVG
          className="svg"
          role="presentation"
        >
          <svg
            className="svg"
            height={16}
            role="presentation"
            viewBox="0 0 16 16"
            width={16}
          >
            <path
              d="M8 7a1 1 0 011 1v3a1 1 0 11-2 0V8a1 1 0 011-1zM8 6a1 1 0 100-2 1 1 0 000 2z"
            />
            <path
              clipRule="evenodd"
              d="M0 8a8 8 0 1116 0A8 8 0 010 8zm8-6a6 6 0 100 12A6 6 0 008 2z"
              fillRule="evenodd"
            />
          </svg>
        </InformationSVG>
      </span>
    </div>
  </div>
</WholesalePrice>
`;

exports[`<WholesalePrice /> should render without props 1`] = `
<WholesalePrice>
  <div
    className=""
  >
    <div
      className="wrapper"
    >
      <ForwardRef(Text)
        color="gray38"
        lineHeight={16}
        size={12}
        weight={600}
      >
        <span
          className="size-12 line-height-16 color__gray38 weight-600"
        >
          <Price
            currencyGraphemeClass="currency-grapheme"
          >
            <span
              className="root"
            />
          </Price>
           / 
          <span
            className="wholesale-word"
          >
            Опт 
          </span>
        </span>
      </ForwardRef(Text)>
    </div>
  </div>
</WholesalePrice>
`;
